### PR TITLE
[NPU] Compatible with other third-party models like auto-round

### PR DESCRIPTION
**Pull request by rnwang04**
### 1. Why the change?

Port from https://github.com/intel-analytics/ipex-llm/pull/12581.
When we  convert other third-party models to our NPU model, take auto-round for example, I will directly convert their `QuantLinear` to our `QuantizedLinear` and will not call `split_linears` in this progress.
So linear of model converted from auto-round will not have `_dq_list` suffix.

### 2. User API changes

No change.

### 3. Summary of the change 

- update our code to be compatible with other third-party models like auto-round
- simplify code
- fix GW sym  int4 runtime error
- code refactor

### 4. How to test?
- [ ] N/A
- [ ] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/npu_pipeline_model/qwen.py [hunk 1]
```diff
@@ -263,31 +278,40 @@ def convert_fused_qwen_layer(model, fused_layers, n_splits_linear, n_splits_down
         k_biases = []
         v_biases = []
         layer_indexs = range(layer_start, layer_end)
-        n_splits_linear = len(model.model.layers[0].mlp.gate_proj_dq_list)
```
Comment (plusbang) [inline python/llm/src/ipex_llm/transformers/npu_pipeline_model/qwen.py:266]: Please carefully confirm whether we could remove this here. Other LGTM

**Review by plusbang — APPROVED** [review]

File: python/llm/src/ipex_llm/transformers/npu_pipeline_model/qwen.py (hunk 1 above)
Comment (rnwang04) [reply python/llm/src/ipex_llm/transformers/npu_pipeline_model/qwen.py:266]: Yes, I have confirmed it with qwen2 GW sym int4.
Remove this won't impact program running or performance, as these two parameters are passed in as inputs.

File: python/llm/src/ipex_llm/transformers/npu_pipeline_model/qwen.py [hunk 2]
```diff
@@ -134,27 +134,42 @@ def convert_qwen_layer(model, layer_idx, n_splits_linear, n_splits_down_proj,
...
-    k_bias = attn_layer.k_proj_dq_list.k_proj_dq_0.bias.to(torch.float16)
-    v_bias = attn_layer.v_proj_dq_list.v_proj_dq_0.bias.to(torch.float16)
+    if hasattr(attn_layer, "q_proj_dq_list"):
+        q_bias = attn_layer.q_proj_dq_list.q_proj_dq_0.bias.to(torch.float16)
+        k_bias = attn_layer.k_proj_dq_list.k_proj_dq_0.bias.to(torch.float16)
+        v_bias = attn_layer.v_proj_dq_list.v_proj_dq_0.bias.to(torch.float16)
+    else:
+        q_bias = attn_layer.q_proj.bias.to(torch.float16)
+        k_bias = attn_layer.k_proj.bias.to(torch.float16)
+        v_bias = attn_layer.v_proj.bias.to(torch.float16)
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/npu_pipeline_model/qwen.py:172]: Shall we refactor the related logics into a common utility method?

Comment (rnwang04) [reply python/llm/src/ipex_llm/transformers/npu_pipeline_model/qwen.py]: done, have refactor related code.

**Review by jason-dai — APPROVED** [review]
LGTM

**Pull request merged**